grobid-core/src/main/java/org/grobid/core/jni/DeLFTModel.java

```diff
@@ -208,7 +210,30 @@ public void run() {
                 LOGGER.error("DeLFT model training via JEP failed", e);
             } 
         } 
-    } 
+    }
+
+    protected static List<String> getTrainCommand(String modelName, File trainingData) {
+        String trainModule = GrobidProperties.getDeLFTTrainModule();
```

Comment:
I think here you should export the ENV variable with the conda environment... or check if it exists and use it when running the command... 

Right now, running it from a submodule, I get:

```
18:18:35.610 [main] INFO org.grobid.core.jni.DeLFTModel - Train DeLFT model superconductors...
18:18:35.610 [main] INFO org.grobid.core.jni.DeLFTModel - Running: [python3, grobidTagger.py, superconductors, train, --input, /Users/lfoppiano/development/projects/grobid/grobid-home/tmp/superconductors3567936683595801467.train, --output, /Users/lfoppiano/development/projects/grobid/grobid-superconductors/../grobid-home/models]
Traceback (most recent call last):
  File "grobidTagger.py", line 3, in <module>
    from delft.utilities.Embeddings import Embeddings
  File "/Users/lfoppiano/development/projects/delft/delft/utilities/Embeddings.py", line 3, in <module>
    from keras.preprocessing import text, sequence
ModuleNotFoundError: No module named 'keras'
```

Reply:
Any though how to fix this? I'm not sure what we can do because by running the command imply we need to load the whole environment

Reply:
Have you tried activating the environment first? I would argue it's outside the scope of this PR.
But as with using JEP in general, personally I would leave activating the environment to the user as trying to activate it programatically with JEP seems less stable.

Reply:
OK sorry if I run it after having activated the environment, it works. 
It doesn't work, unfortunately, using the CONDA env variable, because is not just a matter of libraries but also paths... since the training is ran using python command 😢 

No idea if this can be fixed...

Reply:
> OK sorry if I run it after having activated the environment, it works.
> It doesn't work, unfortunately, using the CONDA env variable, because is not just a matter of libraries but also paths... since the training is ran using python command 
> 
> No idea if this can be fixed...

Not sure if I understand you correctly. Everything works when activating the conda environment first?

Not sure about the CONDA env variable, do you mean `CONDA_PREFIX`?. I guess that is related to the Python Environment config and the virtual env config in `grobid.properties` that you added? When running an external command, it is certainly easier to properly _activate_ a virtual environment as environment variables can be passed in (without hacking some Java internals). You could just add `${CONDA_PREFIX}/bin` to (the beginning of) the `PATH`.

Reply:
I wonder if we could read the CONDA_PREFIX and pick up python from that location instead of just calling a generic `python3` executable... 

In this way we maintain the double compatibility: 
 1. environment activation + command launch and  
 2. environment setting + command launch 

Luca

Reply:
I've created a separate issue for that